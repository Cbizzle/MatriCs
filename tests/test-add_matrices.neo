int main() {
	int[2,2] a;
	int[2,2] b;
	int[4] a_flat;
	int[4] b_flat;
	int[4] d_flat;
	int[2,2] mat;
	
	int i;
	int j;
	int k;
	int l;
	int m;

	int c;
	int c1;
	int c2;

	a = [[1,2], [3,4]];
	b = [[5,6], [7,8]];
	
	c = 0;
	for(i = 0; i < 2; i = i + 1) {
		for (j = 0; j < 2; j = j + 1) {
			a_flat[c] = a[i, j];
			b_flat[c] = b[i, j];
			c = c + 1;
		}
	}
	
	c1 = 0;	
	for(k = 0; k < 4; k = k + 1) {
		d_flat[c1] = a_flat[c1] + b_flat[c1];
		c1 = c1 + 1;
	}
	
	c2 = 0;	
	for(l = 0; l < 2; l = l + 1) {
		for(m = 0; m < 2; m = m + 1) {
			mat[l,m] = d_flat[c2];
			c2 = c2 + 1; 						
		}
	}
	
	print_int(mat[0,0]);
	print_int(mat[0,1]);
	print_int(mat[1,0]);
	print_int(mat[1,1]);
		
	return 0;			
}
