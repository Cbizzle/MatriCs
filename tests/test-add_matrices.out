6
8
10
12
7
